pkg/config/config.go

```diff
@@ -388,3 +422,106 @@ func checkAdvertiseAddressConfigured(advertiseAddress string) error {
 	}
 	return fmt.Errorf("Advertise address: %s not present in any interface", advertiseAddress)
 }
+
+func validateRouterListenAddress(entries []string, advertiseAddress string, skipInterface bool) error {
+	addresses, err := AllowedListeningIPAddresses()
+	if err != nil {
+		return err
+	}
+	names, err := AllowedNICNames()
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		if net.ParseIP(entry) != nil {
+			if entry == advertiseAddress && !skipInterface {
+				continue
+			}
+			if !slices.Contains(addresses, entry) {
+				return fmt.Errorf("IP %v not present in the host", entry)
+			}
+		} else if slices.Contains(names, entry) {
+			continue
+		} else {
+			return fmt.Errorf("entry %v not present in the host", entry)
+		}
+	}
+	return nil
+}
+
+func getBannedIPs() ([]*net.IPNet, error) {
+	banned := make([]*net.IPNet, 0)
+	for _, entry := range defaultRouterForbiddenCIDRs {
+		_, netIP, err := net.ParseCIDR(entry)
+		if err != nil {
+			return nil, err
+		}
+		banned = append(banned, netIP)
+	}
+	return banned, nil
+}
+func getHostAddresses() ([]net.IP, error) {
+	handle, err := netlink.NewHandle()
+	if err != nil {
+		return nil, err
+	}
+	links, err := handle.LinkList()
+	if err != nil {
+		return nil, err
+	}
+	addresses := make([]net.IP, 0)
+	for _, link := range links {
+		if link.Attrs().ParentIndex != 0 || link.Attrs().MasterIndex != 0 {
```

Comment:
What do `ParentIndex` and `MasterIndex` mean? What does this line do?

Reply:
Those variables tell you if the link is a slave to another link (like veth's from pods). Will include a comment.